curtsies/escseqparse.py

```diff
@@ -65,69 +87,103 @@ def peel_off_esc_code(s):
-                (?P<intermed>""" + '[\x20-\x2f]*)' + r"""
-                (?P<command>""" + '[\x40-\x7e]))' + r"""
+                (?P<intermed>"""
+        + "[\x20-\x2f]*)"
+        + r"""
+                (?P<command>"""
+        + "[\x40-\x7e]))"
+        + r"""
             (?P<rest>.*)"""
+    )
     m1 = re.match(p, s, re.VERBOSE)  # multibyte esc seq
-    m2 = re.match('(?P<front>.*?)(?P<seq>(?P<csi>)(?P<command>[\x40-\x5f]))(?P<rest>.*)', s)  # 2 byte escape sequence
+    m2 = re.match(
+        "(?P<front>.*?)(?P<seq>(?P<csi>)(?P<command>[\x40-\x5f]))(?P<rest>.*)",
+        s,
+    )  # 2 byte escape sequence
     m = None  # Optional[Match[str]]
     if m1 and m2:
-        m = m1 if len(m1.groupdict()['front']) <= len(m2.groupdict()['front']) else m2
+        m = (
+            m1
+            if len(m1.groupdict()["front"]) <= len(m2.groupdict()["front"])
+            else m2
+        )
         # choose the match which has less processed text in order to get the
         # first escape sequence
-    elif m1: m = m1
-    elif m2: m = m2
-    else: m = None
+    elif m1:
+        m = m1
+    elif m2:
+        m = m2
+    else:
+        m = None
 
     if m:
         d = m.groupdict()  # type: Dict[str, Any]
-        del d['front']
-        del d['rest']
-        if 'numbers' in d and all(d['numbers'].split(';')):
-            d['numbers'] = [int(x) for x in d['numbers'].split(';')]
+        del d["front"]
+        del d["rest"]
+        if "numbers" in d and all(d["numbers"].split(";")):
+            d["numbers"] = [int(x) for x in d["numbers"].split(";")]
 
-        return m.groupdict()['front'], cast(Token, d), m.groupdict()['rest']
+        return m.groupdict()["front"], cast(Token, d), m.groupdict()["rest"]
     else:
-        return s, None, ''
+        return s, None, ""
+
 
 def token_type(info):
     # type: (Token) -> Optional[List[Dict[Text, Union[Text, bool, None]]]]
     """
     """
-    if info['command'] == 'm':
+    if info["command"] == "m":
         # The default action for ESC[m is to act like ESC[0m
         # Ref: https://en.wikipedia.org/wiki/ANSI_escape_code#CSI_codes
-        values = cast(List[int], info['numbers']) if len(info['numbers']) else [0]
+        values = (
+            cast(List[int], info["numbers"]) if len(info["numbers"]) else [0]
+        )
         tokens = []  # type: List[Dict[str, Union[Text, bool, None]]]
         for value in values:
-            if value in FG_NUMBER_TO_COLOR: tokens.append({'fg':FG_NUMBER_TO_COLOR[value]})
-            if value in BG_NUMBER_TO_COLOR: tokens.append({'bg':BG_NUMBER_TO_COLOR[value]})
-            if value in NUMBER_TO_STYLE: tokens.append({NUMBER_TO_STYLE[value]:True})
-            if value == RESET_ALL: tokens.append(dict(dict((k, None) for k in STYLES), **{'fg':None, 'bg':None}))
-            if value == RESET_FG: tokens.append({'fg':None})
-            if value == RESET_BG: tokens.append({'bg':None})
+            if value in FG_NUMBER_TO_COLOR:
```

Comment:
This one makes me sad too (just the for loop), and looking up the syntax it's easy to ignore https://stackoverflow.com/a/58584557/398212